src/replication.c

```diff
@@ -538,11 +544,13 @@ void replicationFeedStreamFromMasterStream(char *buf, size_t buflen) {
     /* Debugging: this is handy to see the stream sent from master
      * to slaves. Disabled with if(0). */
     if (0) {
-        printf("%zu:",buflen);
-        for (size_t j = 0; j < buflen; j++) {
-            printf("%c", isprint(buf[j]) ? buf[j] : '.');
+        if (!server.hide_user_data_from_log) {
+            printf("%zu:",buflen);
+            for (size_t j = 0; j < buflen; j++) {
+                printf("%c", isprint(buf[j]) ? buf[j] : '.');
+            }
+            printf("\n");
```

Comment:
well, in some cases user don't define a logfile in the config, and the logs go to stdout, and stdout is redirected to a file.
so a serverLog and printf call will be somewhat equivalent.

since this is a disabled debug code, it doesn't really matter. 
i.e. had it been in comment rather than `if(0)` or had it been in `#if 0` instead, would we bother to modify it?

maybe it's time to delete these `if(0)` blocks altogether (if we need them we can re-add some day, without committing).

Reply:
bottom line, i think we should either keep this modification, or delete the whole block.
out of the two, i'd vote for the second one.